inc/inventory/asset/software.class.php

```diff
@@ -522,23 +532,48 @@ private function populateVersions() {
     * @return void
     */
    private function storeSoftwares() {
+      global $DB;
+
       $software = new GSoftware();
-      $options = [];
-      if ($this->disable_unicity_check === true) {
-         $options['disable_unicity_check'] = true;
-      }
+      $soft_fields = $DB->listFields($software->getTable());
+      $stmt = $stmt_types = null;
 
       foreach ($this->data as $val) {
          $skey = $this->getSoftwareKey($val->name, $val->manufacturers_id);
          if (!isset($this->softwares[$skey])) {
-            $softwares_id = $software->add(
-               Toolbox::addslashes_deep((array)$val),
-               $options,
-               false
-            );
+            $stmt_columns = $this->cleanInputToPrepare((array)$val, $soft_fields);
+
+            if ($stmt === null) {
+               $stmt_types = $types = str_repeat('s', count($stmt_columns));
+               $reference = array_replace(
+                  $stmt_columns,
+                  array_fill_keys(
+                     array_keys($stmt_columns),
+                     new QueryParam()
+                  )
+               );
```

Comment:
Maybe I am wrong, but if you replace all the keys of `$stmt_columns`, then you could directly use `array_fill_keys` result:
```suggestion
               $reference = array_fill_keys(
                  array_keys($stmt_columns),
                  new QueryParam()
               );
```

Reply:
Seems ok, I've replaced that